sklearn/metrics/tests/test_common.py

```diff
@@ -1813,6 +1813,33 @@ def check_array_api_multiclass_classification_metric(
     )
 
 
+def check_array_api_regression_metric_1d(metric, array_namespace, device, dtype_name):
```

Comment:
```suggestion
def check_array_api_regression_metric(metric, array_namespace, device, dtype_name):
```

and rename `check_array_api_regression_metric` that was used for `r2_score` as `check_array_api_regression_metric_multioutput` ? 
 
Also, it seems `r2_score` can also be tested with this new check.